### DISCUSSION TITLE
Is it possible to exclude some tests from Allure report?

### DISCUSSION THREAD

**Discussion by lancheg** [allure-framework/allure2#2291] [Ideas & Feedback]
Do we have a way to exclude some tests from allure report, but still run it?

Problem: one test project, have 100+ tests, now all [Fact]/[Theory] goes to report and creates mess.

Before [this changes](https://github.com/allure-framework/allure-csharp/issues/344), it was possible to manually decide and mark some tests with [AllureXunit] plus some allure attributes like AllureId/AllureSuite etc, other tests are just reported to CI/IDE.

Now we are forced to have all 100+ tests in Allure.

**Comment by baev** [comment]
Can we have more context on why you don't want all the tests to be shown in the report?

**Reply by lancheg** [reply]
We use Allure for sharing our tests with QA team. QA writes test-case in Allure, then we link unit test with Allure test case by AllureId, and we want that only such tests appear in the Allure report. All other cases which are not linked with Allure test cases we prefer to stay in our code only without uploading to Allure. Is that possible to do?

This worked right as we need in v2.1.3 with AllureXunit attribute , but after we moved to v2.11.0 we have all the tests uploaded to Allure.

**Reply by baev** [reply]
Just to be clear — are you Allure TestOps user?

**Reply by lancheg** [reply]
Yep, I am an Allure TestOps user

**Comment by OlgaChatalyan** [comment]
Hi lancheg,
Yes, we've made the change, which reduced the ability to add specific tests to the report. Now, all tests from the project are included in the report. We will add the "out" option later (not sure when exactly).
For now, I'd like to propose two ways to exclude tests that should not be included in the report:
Option 1:
Move the tests that you report to Allure to a separate project. Install the Allure.XUnit package only into that project and remove it from the original one.
Option 2:
Use the NoAutoReporters option of [xUnit.net](http://xunit.net/) to disable Allure. Enable it (by omitting NoAutoReporters or applying RunnerSwitch=allure) only with a filter that selects the tests to be reported to TestOps. You can specify the options via the dotnet CLI or a .runsettings file. See more info here: https://xunit.net/docs/runsettings and here: https://xunit.net/docs/runsettings. Also, refer to the documentation for your IDE on how to apply a .runsettings file.
You can use xUnit's [Trait] attribute to mark tests and the --filter argument of the dotnet CLI to run them. For example, to run all tests marked with [Trait("Category", "Allure")] and include them in the report using the dotnet CLI use:
dotnet test --filter Category=Allure
To run the other tests, use (for the most recent version of xunit.runner.visualstudio):
dotnet test --filter Category!=Allure -- xUnit.NoAutoReporters=true
The downside is that you must split your test base into two groups and run them separately.
